taskc/simple.py

```diff
@@ -11,86 +11,148 @@
+        """
+
+        # Build TaskConnection object
+        connection = cls(**kwargs)
+
+        # Read the taskrc file to make a dict of attributes/values
         conf = dict([x.replace("\\/", "/").strip().split('=') for x in open(
-            os.path.expanduser(f)).readlines() if '=' in x and x[0] != "#"])
-        self.client_cert = conf['taskd.certificate']
-        self.client_key = conf['taskd.key']
-        self.server = conf['taskd.server'].split(":")[-2]
-        self.port = int(conf['taskd.server'].split(":")[-1])
-        self.cacert_file = conf['taskd.ca'] if 'taskd.ca' in conf else None
-        self.group, self.username, self.uuid = conf['taskd.credentials'].split("/")
-        return self
+            os.path.expanduser(taskrc)) if '=' in x and x[0] != "#"])
+
+        # Set instance variables according to taskrc file
+        connection.client_cert = conf['taskd.certificate']
+        connection.client_key = conf['taskd.key']
+        connection.server = conf['taskd.server'].split(":")[-2]
+        connection.port = int(conf['taskd.server'].split(":")[-1])
+        connection.cacert_file = conf[
+            'taskd.ca'] if 'taskd.ca' in conf else None
+        connection.group, connection.username, connection.uuid = conf[
+            'taskd.credentials'].split("/")
+
+        return connection
 
     def connect(self):
-        "Actually open the socket"
-        c = ssl.create_default_context()
-        c.load_cert_chain(self.client_cert, keyfile=self.client_key)
+        """
+        Actually open the socket
+        """
+
+        # Initialize an SSL context
+        context = ssl.create_default_context()
+        context.load_cert_chain(self.client_cert, keyfile=self.client_key)
         if self.cacert_file:
-            c.load_verify_locations(cafile=self.cacert_file)
+            context.load_verify_locations(cafile=self.cacert_file)
         elif self.cacert:
-            print self.cacert
-            c.load_verify_locations(cadata=self.cacert)
+            print self.cacert  # TODO: Replace prints with logging
+            context.load_verify_locations(cadata=self.cacert)
+
         # enable for non-selfsigned certs
         # print conn.getpeercert()
-        c.check_hostname = False
-        self.conn = c.wrap_socket(socket.socket(socket.AF_INET))
+        context.check_hostname = False
+
+        # create the socket
+        self.conn = context.wrap_socket(socket.socket(socket.AF_INET))
         self.conn.connect((self.server, self.port))
 
     def recv(self):
-        "Parse out the size header & read the message"
+        """
+        Parse out the size header & read the message
+        """
+
+        # receive data through the socket
         a = self.conn.recv(4)
-        bytez = struct.unpack('>L', a[:4])[0]
-        msg = self.conn.recv(bytez)
+
+        # Read the number of bytes of the message
+        # ">L" is 4 bytes long in big endian
+        bytes = struct.unpack('>L', a[:4])[0]
```

Comment:
It's not. `bytes` is the name of builtin function. When the bytes variable is created, its name shadow the bytes function's name. Thus, you can't use the bytes function in this scope. But it's totaly valide.